src/reporting/Market.sol

```diff
@@ -0,0 +1,604 @@
+            return false;
+        }
+        if (_shadyToken.getTypeName() != "ReportingToken") {
+            return false;
+        }
+        bytes32 _shadyId = _shadyToken.getPayoutDistributionHash();
+        ReportingToken _reportingToken = reportingTokens[_shadyId];
+        if (address(_reportingToken) == NULL_ADDRESS) {
+            return false;
+        }
+        if (_reportingToken != _shadyToken) {
+            return false;
+        }
+        return true;
+    }
+
+    function isContainerForShareToken(IShareToken _shadyShareToken) public constant returns (bool) {
+        if (_shadyShareToken.getTypeName() != "ShareToken") {
+            return false;
+        }
+        return(getShareToken(_shadyShareToken.getOutcome()) == _shadyShareToken);
+    }
+
+    function isContainerForDisputeBondToken(DisputeBondToken _shadyBondToken) public constant returns (bool) {
+        if (_shadyBondToken.getTypeName() != "DisputeBondToken") {
+            return false;
+        }
+        if (automatedReporterDisputeBondToken == _shadyBondToken) {
+            return true;
+        } else if (limitedReportersDisputeBondToken == _shadyBondToken) {
+            return true;
+        } else if (allReportersDisputeBondToken == _shadyBondToken) {
+            return true;
+        }
+        return false;
+    }
+
+    function canBeReportedOn() public constant returns (bool) {
+        // CONSIDER: should we check if migration is necessary here?
+        if (isFinalized()) {
+            return false;
+        }
+        if (!reportingWindow.isReportingActive()) {
+            return false;
+        }
+        return true;
+    }
+
+    function needsMigration() public constant returns (bool) {
+        if (isFinalized()) {
+            return false;
+        }
+        Market _forkingMarket = getBranch().getForkingMarket();
+        if (address(_forkingMarket) == NULL_ADDRESS) {
+            return false;
+        }
+        if (_forkingMarket == this) {
+            return false;
+        }
+        if (block.timestamp < endTime) {
+            return false;
+        }
+        if (automatedReporterAddress != NULL_ADDRESS && block.timestamp < getAutomatedReportDueTimestamp()) {
+            return false;
+        }
+        if (automatedReportReceived && block.timestamp < getAutomatedReportDisputeDueTimestamp()) {
+            return false;
+        }
+        if (automatedReportReceived && address(automatedReporterDisputeBondToken) == NULL_ADDRESS) {
+            return false;
+        }
+        return true;
+    }
+
+    function getAutomatedReportDueTimestamp() public constant returns (uint256) {
+        return endTime + AUTOMATED_REPORTING_DURATION_SECONDS;
+    }
+
+    function getAutomatedReportDisputeDueTimestamp() public constant returns (uint256) {
+        return getAutomatedReportDueTimestamp() + AUTOMATED_REPORTING_DISPUTE_DURATION_SECONDS;
+    }
+
+    function getReportingState() public constant returns (ReportingState) {
```

Comment:
As far as order goes, AWAITING_MIGRATION probably should live between automated reporting and limited reporting?  A market won't migrate until automated reporting/dispute are over and the market would otherwise enter LIMITED_REPORTING.